File: src/leetcode/math/T3876.java
class Solution {
    public boolean uniformArray(int[] nums1) {
        int mn = nums1[0];
        boolean hasOdd = false;
        for (int i = 0; i < nums1.length; i++) {
            if (nums1[i] < mn) {
                mn = nums1[i];
            }
            if ((nums1[i] & 1) == 1) {
                hasOdd = true;
            }
        }    
        if ((mn & 1) == 1) {
            return true;
        }
        return !hasOdd;
    }
}

File: code/math/T3876.py
class Solution:
    def uniformArray(self, nums1: list[int]) -> bool:
        mn = nums1[0]
        hasOdd = False
        for v in nums1:
            if v & 1:
                hasOdd = True
            mn = min(mn, v)
        if mn & 1:
            return True
        return not hasOdd
        
        




        